lib/matplotlib/axes/_base.py

```diff
@@ -1322,6 +1324,15 @@ def axis(self, *v, **kwargs):
                     self.autoscale_view(tight=True)
                     self.set_autoscale_on(False)
                     self.set_aspect('equal', adjustable='box', anchor='C')
+                elif s == 'square':
+                    self.set_aspect('equal', adjustable='box', anchor='C')
+                    self.set_autoscale_on(False)
+                    xlim = self.get_xlim()
+                    ylim = self.get_ylim()
+                    minlim = min(xlim[0], ylim[0])
```

Comment:
Would in be better to change the limits to

```python
edge_size = max(np.diff(xlim), np.diff(ylim))
ax.set_xlim([xliml[0], xlim[0] + edge_size])
ax.set_ylim([ylim[0], ylim[0] + edges_size])
```

Which would let you get a square area, but with the lower left any where is dataspace, not just on the `x==y` diagonal.